tests/unit/aws/handlers/openapi.py

```diff
@@ -0,0 +1,88 @@
+import json
+
+from rolo import Request, Response
+from rolo.gateway import RequestContext
+from rolo.gateway.handlers import EmptyResponseHandler
+
+from localstack.aws.chain import HandlerChain
+from localstack.aws.handlers.openapi import OpenAPIRequestValidator
+
+
+class TestOpenAPIRequestValidator:
+    def test_valid_request(self):
+        chain = HandlerChain([OpenAPIRequestValidator()])
+        context = RequestContext(
+            Request(
+                path="/_localstack/diagnose",
+                method="GET",
+                scheme="http",
+                headers={"Host": "localhost.localstack.cloud:4566"},
```

Comment:
Every successful test has a clear host header here (always using `localhost.localstack.cloud:4566`).
Could you maybe add a test verifying that this is not a necessity? The requests need to work just as fine if the `Host` header contains `localhost:4566` or any other value.